### PR TITLE
refactor: adhere mostly to StandardJS guidelines

### PR DESCRIPTION
**Pull request by jviotti**
This commit changes the whole codebase to adhere to all StandardJS
guidelines rules except semicolons, since the removal of semicolons
affect pretty much all lines, and the final diff is very hard to follow
(and to assess other more involved changes).

In a nutshell:

- When using `function`, we now require a space before the opening
  parenthesis
- If a line with operators is broken into multiple lines, the operator
  should now go after the line break
- Unnecessary padding lines are now forbidden

There were also some minor things that the `standard` CLI caught that I
updated here.

See: https://standardjs.com
Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

**Comment by jhermsmeier** [comment]
What do you think about using [standard/eslint-config-standard](https://github.com/standard/eslint-config-standard) and extend that, to shrink down our custom defined rule set?

**Comment by jviotti** [comment]
Yep, we definitely should. I'll handle it it another PR.

**Comment by lurch** [comment]
Out of curiosity, have any / many other resin.io repos switched to this new style?

**Comment by jviotti** [comment]
I think we're the pioneers :P

File: lib/gui/app.js [hunk 1]
```diff
@@ -297,10 +292,9 @@ app.controller('HeaderController', function(OSOpenExternalService) {
     const supportUrl = selectionState.getImageSupportUrl() || DEFAULT_SUPPORT_URL;
     OSOpenExternalService.open(supportUrl);
   };
-
 });
 
-app.controller('StateController', function($rootScope, $scope) {
+app.controller('StateController', function ($rootScope, $scope) {
```
Comment (lurch) [inline lib/gui/app.js]: Out of curiosity (this may well be another one of my stupid questions) how come this is still using the `function` keyword rather than fat-arrow notation?

File: lib/gui/components/safe-webview.js [hunk 2]
```diff
@@ -78,18 +78,17 @@ const API_VERSION = 1;
...
-  constructor(props) {
+  constructor (props) {
     super(props);
 
     this.state = {
       shouldShow: true
     };
 
-    const url = new URL(props.src);
+    const url = new window.URL(props.src);
```
Comment (lurch) [inline lib/gui/components/safe-webview.js]: @Shou is this correct?

File: lib/gui/app.js (hunk 1 above)
Comment (jviotti) [inline lib/gui/app.js]: Angular relies on being able to set custom contexts with `.call()` and `.apply()`, which fat arrow functions don't support. Basically, Angular.js 2 is not ES6 ready.

**Comment by jhermsmeier** [comment]
> Out of curiosity, have any / many other resin.io repos switched to this new style?

@lurch yup https://github.com/resin-io/resin-preload

File: lib/gui/components/safe-webview.js (hunk 2 above)
Comment (Shou) [inline lib/gui/components/safe-webview.js]: Yup! URL lives on `window`, and this makes it explicit, so a good change.

**Comment by lurch** [comment]
@jhermsmeier Yup, saw that ;-)
